helm-charts/chatqna/values.yaml

```diff
@@ -7,6 +7,14 @@
 
 replicaCount: 1
 
+# Enabling HPA will:
+# - Overwrite existing PrometheusAdapter "adapter-config" configMap with ChatQnA specific custom metric queries
+#   for embedding, reranking, tgi services
+# Upstream default configMap:
+#  - https://github.com/kubernetes-sigs/prometheus-adapter/blob/master/deploy/manifests/config-map.yaml
+horizontalPodAutoscaler:
```

Comment:
Create the switch here and make it default to false is needed. But make customMetrics.yaml working, we need also to turn on the horizontalPodAutoscaler.enabled switch for tgi/tei/teirerank separately.

I would suggest creating a new value files with all the HPA switches settings-on, and add notes in the chatqna README of how to use it. Just like what we did with gaudi-values.

Reply:
Because custom metrics for all 3 common components are currently added in `chatqna`, READMEs for those common components should mention enabling `horizontalPodAutoscaler` in `chatqna` as their Pre-condition.

However, those scaled components are common ones, i.e. this is not ChatQnA specific...  So, what if custom metrics would have its own Helm chart under common/, and the 3 common components would list it as a dependency?

Though best would be if Helm chart for each of the relevant common components could patch the PrometheusAdapter configMap  themselves (add custom metrics / queries related to their HPA rules to it), instead of that configMap being just overwritten. :-/

Reply:
I've moved `enabled` flag to global section, so it propagates to children / dependency charts now.